cilium-cli: Test TLS with serverNames

jrajahalme opened the pull request:
Change the client-egress-l7-tls policy to also require SNI match on the configured external target. Add a new `client-egress-tls-sni` test case to test SNI enforcement without TLS interception. Remove the trailing dot from the external target default value to make this easier.

```release-note
Add coverage for SNI enforcement in cilium-cli connectivity tests.
```

jrajahalme committed:
DO NOT MERGE: Test with updated Envoy image

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

gandro commented on cilium-cli/connectivity/builder/client_egress_tls_sni.go:
```diff
@@ -0,0 +1,33 @@
+		WithScenarios(tests.PodToWorld()).
+		WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+			if a.Destination().Port() == 443 {
+				// && a.Destination().Address(features.GetIPFamily(ct.Params().ExternalTarget)) == ct.Params().ExternalTarget
```
Stray comment? Shouldn't this be enabled?

jrajahalme committed:
ci: Remove trailing dot from 'external-target' parameter

Remove the trailing dot from the external target parameter. It is not
needed, and removing it allows the parameter to be used for SNI
enforcement as-is.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

jrajahalme committed:
cilium-cli: Test TLS with serverNames

Change the client-egress-l7-tls policy to also require SNI match on the
configured external target. Remove the trailing dot from the external
target default value to make this easier.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>
